### PR TITLE
Fix all remaining DCM warnings in `devtools_shared`

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/flutter/devtools/issues/5479

### CODE REVIEW

File: packages/devtools_shared/pubspec.yaml [hunk 1]
```diff
@@ -16,4 +16,5 @@ dependencies:
   webkit_inspection_protocol: '>=0.5.0 <2.0.0'
 
 dev_dependencies:
+  dart_code_metrics: ^5.7.2
   test: ^1.21.2
```
Comment (polina-c) [inline packages/devtools_shared/pubspec.yaml:20]: Can you comment why the dependency is needed?

Comment (elliette) [reply packages/devtools_shared/pubspec.yaml:20]: Hmm having it as a dev dependency is part of the installation instructions, but I don't why it's necessary https://dartcodemetrics.dev/docs/getting-started/installation

I  believe @kenzieschmoll was running into an issue due to not having this as a dev dependency. 

I already added it as a dev dependency to devtools_app and devtools_test

Comment (kenzieschmoll) [reply packages/devtools_shared/pubspec.yaml:20]: From the [instructions](https://dartcodemetrics.dev/docs/cli#available-commands), it was necessary to run commands like: `dart run dart_code_metrics:metrics check-unused-code lib`, `check-unused-files`, etc.

Comment (kenzieschmoll) [reply packages/devtools_shared/pubspec.yaml:20]: but maybe the same can be accomplished by using the `dcm` executable directly.. so maybe it isn't needed after all.

Comment (kenzieschmoll) [reply packages/devtools_shared/pubspec.yaml:20]: @incendial this is something that may need docs updates. That you need _either_ the dev dependency _or_ the dcm executable, but not necessarily both.

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_shared/pubspec.yaml (hunk 1 above)
Comment (incendial) [reply packages/devtools_shared/pubspec.yaml:20]: You're referring to the old website, please use the new one instead (https://dcm.dev/docs/teams/getting-started/ and https://dcm.dev/docs/teams/cli/).

There are two versions available: one is a dart package and requires the dependency to be listed and another one is a standalone executable (former is the "Individuals" version and the latter is "Teams). You have a Teams version, so no dependency is needed.

In our docs they are also separated:
Screenshot 2023-04-20 at 11 08 13
Screenshot 2023-04-20 at 11 08 39

And the url is different.

Hope this helps resolve the misunderstanding.

**Pull request merged**